osbenchmark/synthetic_data_generator/synthetic_data_generator.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import logging
+
+import json
+import os
+
+from dask.distributed import Client
+
+from osbenchmark.utils import console
+from osbenchmark.synthetic_data_generator.input_processor import create_sdg_config_from_args, use_custom_synthetic_data_generator, use_mappings_synthetic_data_generator
+from osbenchmark.synthetic_data_generator.helpers import load_config, write_record_and_publish_summary_to_console
+from osbenchmark.synthetic_data_generator import custom_synthetic_data_generator, mapping_synthetic_data_generator
+
+def orchestrate_data_generation_for_custom_synthetic_data_generator(cfg, sdg_config, custom_config, dask_client):
+    logger = logging.getLogger(__name__)
+    logger.info("Generating data with custom synthetic data generator")
+
+    if cfg.opts("synthetic_data_generator", "test_document"):
+        custom_module = custom_synthetic_data_generator.load_user_module(sdg_config.custom_module_path)
+        generate_fake_document = custom_module.generate_fake_document
+        custom_module_components = custom_config.get('CustomSyntheticDataGenerator', {})
+        logger.info("Custom module components %s", custom_module_components)
+
+        custom_lists = custom_module_components.get('custom_lists', {})
+        custom_providers = {name: getattr(custom_module, name) for name in custom_module_components.get('custom_providers', [])}
+        logger.info("Custom lists %s custom providers %s", custom_lists, custom_providers)
+        document = custom_synthetic_data_generator.generate_test_document(generate_fake_document, custom_lists, custom_providers)
+
+        console.println("Generating a single test document:")
+        console.println("Please verify that the output is generated as intended. \n")
+        print(json.dumps(document, indent=2))
+
+    else:
+        # Generate all documents
+        custom_module = custom_synthetic_data_generator.load_user_module(sdg_config.custom_module_path)
+
+        total_time_to_generate_dataset, generated_dataset_details = custom_synthetic_data_generator.generate_dataset_with_user_module(
+            dask_client, sdg_config,
+            custom_module,
+            custom_config
+        )
+
+        write_record_and_publish_summary_to_console(sdg_config, total_time_to_generate_dataset, generated_dataset_details)
+
+def orchestrate_data_generation_for_mapping_synthetic_data_generator(cfg, sdg_config, dask_client):
```

Comment:
thoughts on using Classes instead of methods to better readability and easy extension in future?

Reply:
I'm open to either route. I initially started with classes but decided to go for functions for the first iteration. 

After this PR gets merged in, I can open a separate PR that solely focuses on restructuring this for cosmetic purposes.

Reply:
https://github.com/opensearch-project/opensearch-benchmark/issues/833